test/extra/NotificationTest.cpp

```diff
@@ -22,16 +31,103 @@ class TestFixture : public testing::Test {
-	TestFixture() = default;
-	~TestFixture() = default;
+	NotificationTest() = default;
+	~NotificationTest() = default;
 
 	// Run once per execution of the test application.
 	// Used only for global setup outside of tests.
 	static void SetUpTestSuite() {}
 	static void TearDownTestSuite() {}
+
+	[[nodiscard]] uprotocol::v1::UUri buildValidTestTopic() const;
+	[[nodiscard]] uprotocol::v1::UUri buildValidDefaultSourceURI() const;
 };
 
-// TODO replace
-TEST_F(TestFixture, SomeTestName) {}
+[[nodiscard]] uprotocol::v1::UUri NotificationTest::buildValidDefaultSourceURI()
+    const {
+	uprotocol::v1::UUri testDefaultSourceURI_;
+	testDefaultSourceURI_.set_authority_name("10.0.0.1");
+	testDefaultSourceURI_.set_ue_id(0x18000);
+	testDefaultSourceURI_.set_ue_version_major(0x1);
+	testDefaultSourceURI_.set_resource_id(0x0);
+	return testDefaultSourceURI_;
+}
+
+[[nodiscard]] uprotocol::v1::UUri NotificationTest::buildValidTestTopic()
+    const {
+	uprotocol::v1::UUri testTopic_;
+	testTopic_.set_authority_name("10.0.0.2");
+	testTopic_.set_ue_id(0x00011101);
+	testTopic_.set_ue_version_major(0xF8);
+	testTopic_.set_resource_id(0x8101);
+	return testTopic_;
+}
+
+TEST_F(NotificationTest, NotificationSuccess) {
+	// Initialize
+	uprotocol::v1::UPayloadFormat format_ =
+	    uprotocol::v1::UPayloadFormat::UPAYLOAD_FORMAT_TEXT;
+	std::optional<uprotocol::v1::UPriority> priority_ =
+	    uprotocol::v1::UPriority::UPRIORITY_CS1;
+	;
+	std::optional<std::chrono::milliseconds> ttl_ =
+	    std::chrono::milliseconds(1000);
+	uprotocol::v1::UUri testDefaultSourceURI_ = buildValidDefaultSourceURI();
+	uprotocol::v1::UUri testTopic_ = buildValidTestTopic();
+
+	// Notify Sink
+	auto transportMockNotificationSink =
+	    std::make_shared<uprotocol::test::UTransportMock>(
+	        testDefaultSourceURI_);
+
+	uprotocol::v1::UMessage capture_msg_;
+	auto callback = [&capture_msg_](const auto& message) {
+		capture_msg_ = message;
+	};
+
+	std::optional<uprotocol::v1::UUri> source_filter;
+	source_filter = testDefaultSourceURI_;
+
+	auto result =
+	    NotificationSink::create(transportMockNotificationSink, testTopic_,
+	                             std::move(callback), std::move(source_filter));
+
+	// Notify Source
+	std::string testPayloadStr = "test_payload";
+	auto transportMockNotificationSource_ =
+	    std::make_shared<uprotocol::test::UTransportMock>(
+	        testDefaultSourceURI_);
+
+	auto movableTopic = testTopic_;
+
+	NotificationSource notificationSource(
+	    transportMockNotificationSource_, std::move(movableTopic),
+	    std::move(testDefaultSourceURI_), format_, priority_, ttl_);
+
+	uprotocol::datamodel::builder::Payload testPayload(testPayloadStr, format_);
+	auto status = notificationSource.notify(std::move(testPayload));
+
+	EXPECT_EQ(
+	    AsString::serialize(
+	        transportMockNotificationSource_->message_.attributes().source()),
+	    AsString::serialize(transportMockNotificationSink->sink_filter_));
```

Comment:
Shouldn't the NotificationSource message's source match the NotificationSink listener's source filter?

Reply:
Done